fauna/v10/queries/deploy.js

```diff
@@ -1,5 +1,79 @@
 const { fql } = require("fauna");
 
+/**
+ * Constructs an FQL Query to create/update a collection according to the passed parameters.
+ *
+ * @param params The params to pass to the create or update call.
+ * @param preview A boolean indicating whether to preview or effect changes.
+ * @returns A Query. The query returns an array of results:
+ *          [{ type: "Collection", name: str, action: "created" | "updated", preview: bool}, ...]
+ */
+const createUpdateCollection = (params, preview = false) => {
+  return fql`
+  {
+    let deleteKey = (key, obj) => {
+      Object.fromEntries(Object.entries(obj).where(e => key != e[0]))
+    }
+    
+    let deleteIndexStatuses = (indexes) => {
+      Object.fromEntries(Object.entries(indexes).map(e => [e[0], deleteKey("status", e[1])]))
+    }
+    
+    let deleteConstraintStatuses = (constraints) => {
+      constraints.map(c => deleteKey("status", c))
+    }
```

Comment:
lol you were not kidding, `Object.remove` would be really nice to have. This is great feedback though - seeing fqlx in action is very cool.